src/qamqpmessage.cpp

```diff
@@ -108,6 +108,11 @@ QVariant QAmqpMessage::header(const QString &header, const QVariant &defaultValu
     return d->headers.value(header, defaultValue);
 }
 
+const QHash<QString, QVariant>& QAmqpMessage::headers() const
```

Comment:
`const QHash<QString, QVariant> QAmqpMessage::headers() const` is sufficient since the container is implicitly shared